src/mds/Server.cc

```diff
@@ -1085,9 +1085,13 @@ void Server::recover_filelocks(CInode *in, bufferlist locks, int64_t client)
  */
 void Server::recall_client_state(float ratio)
 {
-  int max_caps_per_client = (int)(g_conf->mds_cache_size * .8);
-  int min_caps_per_client = 100;
-
+  int max_caps_per_client = (int)(g_conf->mds_cache_size * g_conf->max_ratio_caps_per_client);
+  int min_caps_per_client = (int)(g_conf->min_caps_per_client);
+  if (max_caps_per_client < min_caps_per_client) {
+    max_caps_per_client = min_caps_per_client + 1;
+    derr << "max_caps_per_client " << max_caps_per_client << "<" << "min_caps_per_client " << min_caps_per_client << dendl;
```

Comment:
Please change this to a normal debug message that prints **before** you change `max_caps_per_client`.